sdk/playwrighttesting/create-microsoft-playwright-testing/src/initialize.ts

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import prompts, { PromptObject } from "prompts";
+import fs from "fs";
+import { Extensions, Languages, Messages } from "./constants";
+import { OverridePromptResponse, PackageManager, PlaywrightServiceInitConfig } from "./types";
+import { executeCommand, getFileReferenceForImport } from "./utils";
+import { getPackageManager } from "./packageManager";
+
+const questions: PromptObject[] = [
+  {
+    type: "confirm",
+    name: "canOverride",
+    message: Messages.CAN_OVERRIDE_MESSAGE,
+    initial: true,
+  },
+  {
+    type: (prev: boolean) => (prev ? null : "confirm"),
+    name: "confirmationForExit",
+    message: Messages.CONFIRMATION_FOR_EXIT_MESSAGE,
+    initial: true,
+  },
+];
+
+export class PlaywrightServiceInitialize {
+  private _setupConfig: PlaywrightServiceInitConfig;
+  private _packageManager: PackageManager;
+
+  constructor(setupConfig: PlaywrightServiceInitConfig) {
+    this._setupConfig = setupConfig;
+    this._packageManager = getPackageManager();
+  }
+
+  public addServiceSupportToTestSuite = async (): Promise<void> => {
+    const canProceedWithServiceInitialization = await this.checkIfServiceConfigCanBeAdded(); // if service config already present, ask user for overwrite permission
+    if (!canProceedWithServiceInitialization) return;
+    await this.installServicePackage(); // install service packages
+    await this.createServiceConfig(); // create service config file
+    this.displayAdditionalInformation(); // display additional information
+  };
+
+  private checkIfServiceConfigCanBeAdded = async (): Promise<boolean> => {
+    if (!this.isServiceConfigFileAlreadyPresent()) return true;
+    const response = (await prompts.prompt(questions, {
+      onCancel: this.promptOnCancel,
+    })) as OverridePromptResponse;
+    console.log();
+    if (response.canOverride) return true;
+    // eslint-disable-next-line no-return-await
+    if (!response.confirmationForExit) return await this.checkIfServiceConfigCanBeAdded();
+
+    console.log(Messages.SETUP_PROCESS_EXIT_MESSAGE);
+    return false;
+  };
+
+  private promptOnCancel = (): never => {
+    process.exit(0);
+  };
+
+  private isServiceConfigFileAlreadyPresent = (): boolean => {
+    return fs.existsSync(this.getServiceConfigFileName());
+  };
+
+  private displayAdditionalInformation = (): void => {
+    const runCommand = this._packageManager.runCommand(
+      "playwright",
+      `test -c ${this.getServiceConfigFileName()}`,
+    );
+    const runCommandParallelWorkers = this._packageManager.runCommand(
+      "playwright",
+      `test -c ${this.getServiceConfigFileName()} --workers=20`,
+    );
+
+    console.log();
```

Comment:
nit: all of these can be combined into a single `console.log` since strings in backticks will preserve newlines and whitespace